CT-1847 Update visibility of Overturned SAR cases

stephenrichards opened the pull request:
Overturned SAR cases can be seen only by managers, and responders on the
team to which it has been assigned.

beccapearce commented on app/controllers/cases_controller.rb:
```diff
@@ -305,6 +308,33 @@ def execute_request_amends
+    new_external_deadline = Date.new pit_params[:external_deadline_yyyy].to_i,
+                                     pit_params[:external_deadline_mm].to_i,
+                                     pit_params[:external_deadline_dd].to_i
+    cefps = CaseExtendForPITService.new current_user,
```
another name might make this easier to understand like calling it service

misaka replied:
Good call, will change it to `service`.

beccapearce commented on app/models/case_transition.rb:
```diff
@@ -25,8 +25,9 @@ class CaseTransition < ActiveRecord::Base
-    filenames:          [:string, array: true, default: []]
+                 message:        :text,
+                 filenames:      [:string, array: true, default: []],
+                 final_deadline: :text
```
Should this be text or a date?

misaka replied:
Yup! Another good catch!

misaka commented:
@beccapearce I've addressed your comments, care to re-review? :)

misaka commented:
bloody butter-fingers

aliuk2012 commented on app/policies/cases/base_policy.rb:
```diff
@@ -397,6 +404,10 @@ def responder_attachable?
     end
 
+    check :user_is_a_manager_for_case do
+      user.in? self.case.managing_team.users
```
Do we need this? we already have a  check_user_is_a_manager

aliuk2012 replied:
we also have a check_user_is_assigned_manager_for_case i didn't think we assigned managers (BMT) to cases

misaka replied:
I think we do need this:

* `check_user_is_a_manager` -- checks that the user has a managing role with any team
* `check_user_is_a_manager_for_case` -- checks that the user is on the managing team assigned to the case
* `check_user_is_assigned_manager_for_case` -- checks that the manager has been assigned to the case, this is left-over from TMMs and we don't assign users to cases outside of the TMM work that was done. This could be deprecated but that's not the issue here.

aliuk2012 commented on app/services/case_extend_for_pit_service.rb:
```diff
@@ -0,0 +1,25 @@
+    end
+    @result
+  rescue => err
+    Rails.logger.error err.to_s
```
should we be retrofitting these logger methods to our services?

misaka replied:
Ya, so in spirit I like this pattern, but I'm not 100% sure about retrofitting as part of this PR. I'd be in favour to change them as we touch the other services, but I wouldn't want this PR to be about that.

misaka replied:
I think it'd also be nice to have this working for a tiny bit to see what we think of it.

aliuk2012 commented on app/views/cases/show.html.slim:
```diff
@@ -71,6 +71,9 @@ div id="case-#{@case.id}"
 
+  - if policy(@case).execute_extend_for_pit?
+    section.case-info
+      = action_button_for(:extend_for_pit)
```
I cant help wondering why some of our actions use this `action_button_for` helper methods and other dont like line 78-79 below.

misaka replied:
I noticed that too ... I'm not sure why we don't have that wrapped up in a helper, I do think it'd make sense to move that though. But, again, I'm a bit loathe to make that part of this PR though. Especially since it's just the one line below ... I think there is a bit of a bigger job coming to refactor out `@permitted_events`, this could be part of that.

stephenrichards committed:
CT-1847 Update visibility of Overturned SAR cases

Overturned SAR cases can be seen only by managers, and responders on the
team to which it has been assigned.

stephenrichards committed:
CT-1847 make testing utility method #identifier private
